File: find_the_difference_of_two_arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        std::set<int> set1(nums1.begin(), nums1.end());
        std::set<int> set2(nums2.begin(), nums2.end());

        std::set<int> diff1(set1);
        for (int num : set2) {
            diff1.erase(num);
        }

        std::set<int> diff2(set2);
        for (int num : set1) {
            diff2.erase(num);
        }

        std::vector<std::vector<int>> result;
        result.push_back(std::vector<int>(diff1.begin(), diff1.end()));
        result.push_back(std::vector<int>(diff2.begin(), diff2.end()));

        return result;       
    }
};

File: find_the_difference_of_two_arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        Set<Integer> set2 = new HashSet<>();

        for (int num : nums1) {
            set1.add(num);
        }

        for (int num : nums2) {
            set2.add(num);
        }

        Set<Integer> diff1 = new HashSet<>(set1);
        diff1.removeAll(set2);

        Set<Integer> diff2 = new HashSet<>(set2);
        diff2.removeAll(set1);

        List<List<Integer>> result = new ArrayList<>();
        result.add(new ArrayList<>(diff1));
        result.add(new ArrayList<>(diff2));

        return result;
    }
}
